src/TS/services/runInfoDomManipulation.ts

```diff
@@ -1,85 +1,107 @@
+  startCallback: () => void,
+) {
+  const startButton = document.querySelector('#start-btn');
+  if (!startButton) {
+    throw new Error('Seems like an error from our side!');
   }
+  startButton.addEventListener('click', startCallback);
+
+  const lapList = document.querySelector('.laps');
 
-  let lapTime = 'DNF';
-  if (lapSummary['time lap']) {
-    lapTime = new Date(lapSummary['time lap']).toISOString().slice(11, 19);
+  if (!lapList) {
+    throw new Error('Seems like an error from our side!');
   }
 
-  let avg: string = 'N/A';
-  if (lapSummary['Min Speed GPS'] && lapSummary['Max Speed GPS']) {
-    avg =
-      (lapSummary['Min Speed GPS'] + lapSummary['Max Speed GPS']) / 2 + 'km/h';
-    avg = `${parseInt(avg) / 10} km/h`;
+  if (!lapSummaries?.[0]) {
+    throw new Error('Seems like an error from our side!');
   }
 
-  lapInfo.innerHTML = `
-        <div class="lap-detail lap-time">
-        <span>
-            <i class="fa fa-clock-o" aria-hidden="true"></i>
-            Lap time:
-        </span>
-        ${lapTime}
-        </div>
-        <div class="lap-detail avg-speed">
-            <span>
-                <i class="fa fa-signal" aria-hidden="true"></i>
-                Avg Speed:
-            </span>
-            ${avg}
-        </div>`;
+  let index = 0;
+  for (const lap of lapSummaries) {
+    const lapInfoItem = document.createElement('li');
+
+    const input = document.createElement('input');
+    input.setAttribute('type', 'checkbox');
+    input.setAttribute('id', `${index + 1}`);
+    input.setAttribute('value', `${index + 1}`);
+
+    const label = document.createElement('label');
+
+    let lapTime = 'DNF';
+    if (lap['time lap']) {
+      lapTime = new Date(lap['time lap']).toISOString().slice(11, 19);
+    }
+
+    let avg: string = 'N/A';
+    if (lap['Min Speed GPS'] && lap['Max Speed GPS']) {
+      avg = (lap['Min Speed GPS'] + lap['Max Speed GPS']) / 2 + 'km/h';
+      avg = `${parseInt(avg) / 10} km/h`;
+    }
+
+    label.innerHTML = `
+      <p>${index}</p>
+      <span class="time">
+          <i class="fa fa-clock-o" aria-hidden="true"></i>
+          <p>${lapTime}</p>
+      </span>
+      <span class="speed">
+          <i class="fa fa-signal" aria-hidden="true"></i>
+          <p>${avg}</p>
+      </span>
+     `;
+
+    lapInfoItem.appendChild(input);
+    lapInfoItem.appendChild(label);
+
+    lapList.appendChild(lapInfoItem);
+    index++;
+  }
 }
 
-export function getLapNumber() {
-  const select: HTMLSelectElement | null = document.querySelector('select');
+export function getCheckedLaps() {
+  const lapList = document.querySelectorAll('input');
 
-  if (select) {
-    const option = select.options[select.selectedIndex];
-    if (!isNaN(Number(option.value))) return Number(option.value);
+  if (!lapList) {
+    throw new Error('Seems like an error from our side!');
+  }
+
+  const lapArray: number[] = [];
+  for (const lap of lapList) {
+    if (!lap.checked) {
```

Comment:
Curly braces aren't really needed if the statement after the **if-statement** is only one line (i.e. ` continue `). It's not really important but just something to consider